### PR TITLE
Issue#102 configure home server generator module

### PR DESCRIPTION
**Pull request by Abhai2016**
dev
## ZenHub

* [Main ZenHub ticket](https://app.zenhub.com/workspaces/home-project-5f7b77ff8db73a001cb17009)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

**Review by Romanov-Niko — APPROVED** [review]

File: home-application/src/main/java/com/softserveinc/ita/homeproject/application/apiservice/UserApiImpl.java [hunk 1]
```diff
@@ -1,6 +1,6 @@
 package com.softserveinc.ita.homeproject.application.apiservice;
```
Comment (kh0ma) [inline home-application/src/main/java/com/softserveinc/ita/homeproject/application/apiservice/UserApiImpl.java]: Let's rename the package to `api`. So it will be `com.softserveinc.ita.homeproject.application.api`

File: home-clients/pom.xml [hunk 2]
```diff
@@ -72,8 +72,6 @@
                             <generateApiTests>false</generateApiTests>
                             <generateModelTests>false</generateModelTests>
                             <configOptions>
-                                <java8>true</java8>
```
Comment (kh0ma) [inline home-clients/pom.xml]: Why did you remove it? Can it affect anything?

File: home-clients/pom.xml [hunk 3]
```diff
@@ -72,8 +72,6 @@
...
-                                <dateLibrary>java8</dateLibrary>
```
Comment (kh0ma) [inline home-clients/pom.xml]: Why did you remove it? Can it affect anything?

File: home-server-generator/src/main/resources/mustaches/server/pom.mustache [hunk 4]
```diff
@@ -174,32 +87,13 @@
...
-      <url>https://oss.sonatype.org/content/repositories/snapshots</url>
-      <snapshots>
-        <enabled>true</enabled>
-      </snapshots>
-    </repository>
-  </repositories>
+
   <properties>
-    <java.version>{{#java8}}1.8{{/java8}}{{^java8}}1.7{{/java8}}</java.version>
+    <java.version>11</java.version>
```
Comment (kh0ma) [inline home-server-generator/src/main/resources/mustaches/server/pom.mustache]: Do we really need change it?

File: home-server-generator/pom.xml [hunk 5]
```diff
@@ -13,7 +13,7 @@
 
     <properties>
         <yaml.name>openapi.yaml</yaml.name>
-        <generation.files>OffsetDateTimeProvider.java,pom.xml</generation.files>
+        <generation.files>pom.xml</generation.files>
```
Comment (kh0ma) [inline home-server-generator/pom.xml]: What would be if we just generate all supporting files now?

**Review by kh0ma — CHANGES_REQUESTED** [review]

**Review by alexorlenko — APPROVED** [review]

**Review by Dandria1802 — APPROVED** [review]

File: home-application/src/main/java/com/softserveinc/ita/homeproject/application/apiservice/UserApiImpl.java (hunk 1 above)
Comment (Abhai2016) [reply home-application/src/main/java/com/softserveinc/ita/homeproject/application/apiservice/UserApiImpl.java]: Done.

File: home-clients/pom.xml (hunk 2 above)
Comment (Abhai2016) [reply home-clients/pom.xml]: Well, old team said that generator doesn't work with Java 11 so they had to set java 8. But now we changed generator and it works with new java so we don't have to it anymore.

File: home-clients/pom.xml (hunk 3 above)
Comment (Abhai2016) [reply home-clients/pom.xml]: DateLibrary got back, because it needs for jdk8+.

File: home-server-generator/src/main/resources/mustaches/server/pom.mustache (hunk 4 above)
Comment (Abhai2016) [reply home-server-generator/src/main/resources/mustaches/server/pom.mustache]: I guess because now it works with java 11 so I specified it.

File: home-server-generator/pom.xml (hunk 5 above)
Comment (Abhai2016) [reply home-server-generator/pom.xml]: Not sure if I understood what you mean. If you are about Offset, we discussed that we don't need generation time so pom.xml only left. Without generation files doesn't work at all because I have to specify newer maven plugin in pom.

**Review by PoLytvy — APPROVED** [review]

File: home-clients/pom.xml (hunk 2 above)
Comment (kh0ma) [reply home-clients/pom.xml]: We can depend on libraries that gte java v1.5. Currently, the open API generator doesn't support java 11. I think there are some circumstances that can be broken if we just bump the java version. Let's leave it as it is originally supported by the generator.

File: home-clients/pom.xml (hunk 3 above)
Comment (kh0ma) [reply home-clients/pom.xml]: I didn't get it. Are you saying that you are going to ever this change?

File: home-server-generator/pom.xml (hunk 5 above)
Comment (kh0ma) [reply home-server-generator/pom.xml]: Please look at the doc https://openapi-generator.tech/docs/generators/jaxrs-spec. There is no such option as  `supportingFilesToGenerate`. Please remove.

File: home-server-generator/src/main/resources/mustaches/server/pom.mustache (hunk 4 above)
Comment (kh0ma) [reply home-server-generator/src/main/resources/mustaches/server/pom.mustache]: As I mentioned above we can depend on libraries with java gte v1.5. And open API generator doesn't support java 11. So it would be better to leave it as is and generate only code for java 8.

**Review by kh0ma — CHANGES_REQUESTED** [review]

**Review by OlTym — APPROVED** [review]

File: home-clients/pom.xml (hunk 2 above)
Comment (Abhai2016) [reply home-clients/pom.xml]: Done.

File: home-clients/pom.xml (hunk 3 above)
Comment (Abhai2016) [reply home-clients/pom.xml]: No. I already did.

File: home-server-generator/src/main/resources/mustaches/server/pom.mustache (hunk 4 above)
Comment (Abhai2016) [reply home-server-generator/src/main/resources/mustaches/server/pom.mustache]: Got back 1.8 in java.version.

File: home-server-generator/pom.xml (hunk 5 above)
Comment (Abhai2016) [reply home-server-generator/pom.xml]: Deleted.

**Review by kh0ma — APPROVED** [review]
